qiskit/circuit/commutation_checker.py

```diff
@@ -37,32 +38,21 @@ class CommutationChecker:
     evicting from the cache less useful entries, etc.
     """
 
-    def __init__(self):
+    def __init__(self, standard_gate_commutations: dict = None, cache_max_entries: int = 10**6):
         super().__init__()
-        self.cache = {}
-
-    def _hashable_parameters(self, params):
-        """Convert the parameters of a gate into a hashable format for lookup in a dictionary.
-
-        This aims to be fast in common cases, and is not intended to work outside of the lifetime of a
-        single commutation pass; it does not handle mutable state correctly if the state is actually
-        changed."""
-        try:
-            hash(params)
-            return params
-        except TypeError:
-            pass
-        if isinstance(params, (list, tuple)):
-            return tuple(self._hashable_parameters(x) for x in params)
-        if isinstance(params, np.ndarray):
-            # We trust that the arrays will not be mutated during the commutation pass, since nothing
-            # would work if they were anyway. Using the id can potentially cause some additional cache
-            # misses if two UnitaryGate instances are being compared that have been separately
-            # constructed to have the same underlying matrix, but in practice the cost of string-ifying
-            # the matrix to get a cache key is far more expensive than just doing a small matmul.
-            return (np.ndarray, id(params))
-        # Catch anything else with a slow conversion.
-        return ("fallback", str(params))
+        if standard_gate_commutations is None:
+            self._standard_commutations = {}
+        else:
+            self._standard_commutations = standard_gate_commutations
+        self._cache_max_entries = cache_max_entries
+        """
+            self._cached_commutation has the same structure as standard_gate_commutations, i.e. a
+            dict[pair of gate names][relative placement][tuple of gate parameters] := True/False
+        """
+        self._cached_commutations = {}
+        self._current_cache_entries = 0
+        self._cache_miss = 0
+        self._cache_hit = 0
 
```

Comment:
Are `_cache_miss` and `_cache_hit` used anywhere?

Reply:
I found it useful to have these fields for validating the commutation checker. I can remove it if you think that would improve the code but I would prefer to keep them. So far these two fields are only used in `TestCommutationChecker`